src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -1413,6 +1420,8 @@ internal partial class BoundDynamicIndexerAccess
     {
         protected override OperationKind ExpressionKind => OperationKind.None;
 
+        protected override ImmutableArray<IOperation> Children => this.Arguments.Add(this.ReceiverOpt).As<IOperation>();
```

Comment:
Given that the API is only internal, and the OperationWalker handles null child nodes inside Children, I felt it is better to add all nodes and the client handle null cases.